### PR TITLE
add java solution to cf782b

### PR DESCRIPTION
**Pull request by 876pol**
_If any of the below don't apply to this Pull Request, mark the checkbox as done._

- [x] I have tested my code.
- [x] I have added my solution according to the steps [here](https://usaco.guide/general/adding-solution#steps).
- [x] I have followed the code conventions mentioned [here](https://usaco.guide/general/adding-solution/#code-conventions).
  - I understand that if it is clear that I have not attempted to follow these conventions, my PR will be closed.
  - If changes are requested, I will re-request a review after addressing them.

### CODE REVIEW

File: solutions/silver/cf-782b.mdx [hunk 1]
```diff
@@ -98,6 +98,92 @@ int main() {
...
+	}
+
+	public static void main(String[] args) {
+		Scanner scanner = new Scanner(System.in);
+
+		int n = scanner.nextInt();
+
+		double low = 0;
+		double high =
+		    0;  // high will be the location of the farthest north friend
```
Comment (SansPapyrus683) [inline solutions/silver/cf-782b.mdx:155]: put the comment in a separate line above the variable

File: solutions/silver/cf-782b.mdx [hunk 2]
```diff
@@ -98,6 +98,92 @@ int main() {
...
+	static ArrayList<Double> locations = new ArrayList<>();
+	static ArrayList<Double> speeds = new ArrayList<>();
+
+	static double min_ans = Double.MAX_VALUE;
+	static double curr_min_time;
+	static boolean left_has_max = false;  // which side gives us max answer
+	static boolean right_has_max = false;
+	static double MAX_ERROR = 10e-7;
+
+	/*
```
Comment (SansPapyrus683) [inline solutions/silver/cf-782b.mdx:116]:
```suggestion
	/**
```

File: solutions/silver/cf-782b.mdx [hunk 3]
```diff
@@ -98,6 +98,92 @@ int main() {
...
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.util.*;
+
+public class Solution {
+	static ArrayList<Double> locations = new ArrayList<>();
+	static ArrayList<Double> speeds = new ArrayList<>();
```
Comment (SansPapyrus683) [inline solutions/silver/cf-782b.mdx:108]:
```suggestion
	static List<Double> locations = new ArrayList<>();
	static List<Double> speeds = new ArrayList<>();
```

File: solutions/silver/cf-782b.mdx [hunk 4]
```diff
@@ -98,6 +98,92 @@ int main() {
 
 </CPPSection>
 
+<JavaSection>
+
+```java
+import java.util.*;
+
+public class Solution {
```
Comment (SansPapyrus683) [inline solutions/silver/cf-782b.mdx:106]:
```suggestion
public class MeetingPlace {
```

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Comment by SansPapyrus683** [comment]
deal

**Comment by 876pol** [comment]
😭

File: solutions/silver/cf-782b.mdx [hunk 5]
```diff
@@ -98,6 +100,92 @@ int main() {
...
+	 * time interval.
+	 */
+	public static double get_time(double loc) {
+		double max_time = 0;
+		for (int i = 0; i < locations.size(); i++) {
+			if (locations.get(i) == loc) { continue; }
+			double i_time = Math.abs(locations.get(i) - loc) / speeds.get(i);
+			if (i_time > max_time) {
+				left_has_max =
+				    false;  // reset to false, as we have a new maximum
```
Comment (SansPapyrus683) [inline solutions/silver/cf-782b.mdx:129]: same here

File: solutions/silver/cf-782b.mdx [hunk 6]
```diff
@@ -98,6 +100,92 @@ int main() {
...
+
+public class MeetingPlace {
+	static List<Double> locations = new ArrayList<>();
+	static List<Double> speeds = new ArrayList<>();
+
+	static double min_ans = Double.MAX_VALUE;
+	static double curr_min_time;
+	static boolean left_has_max = false;  // which side gives us max answer
+	static boolean right_has_max = false;
+	static double MAX_ERROR = 10e-7;
```
Comment (SansPapyrus683) [inline solutions/silver/cf-782b.mdx:116]: put this at the top

Comment (SansPapyrus683) [reply solutions/silver/cf-782b.mdx:116]: also declare as final

File: solutions/silver/cf-782b.mdx [hunk 7]
```diff
@@ -98,6 +100,92 @@ int main() {
...
+				} else {
+					left_has_max = true;
+				}
+			}
+		}
+		return max_time;
+	}
+
+	public static void main(String[] args) {
+		Scanner scanner = new Scanner(System.in);
```
Comment (SansPapyrus683) [inline solutions/silver/cf-782b.mdx:151]:
```suggestion
```

File: solutions/silver/cf-782b.mdx [hunk 8]
```diff
@@ -98,6 +100,92 @@ int main() {
...
+
+		int n = scanner.nextInt();
+
+		double low = 0;
+		// high will be the location of the farthest north friend
+		double high = 0;
+		double mid;
```
Comment (SansPapyrus683) [inline solutions/silver/cf-782b.mdx:158]: why not declare this in the loop

**Review by SansPapyrus683 — CHANGES_REQUESTED** [review]

**Review by SansPapyrus683 — APPROVED** [review]

**Review by envyaims — APPROVED** [review]
ok

**Pull request merged**